src/compiler/checker.ts

```diff
@@ -5481,6 +5481,12 @@ namespace ts {
                         }
                     }
                     else {
+                        if (getObjectFlags(type) & ObjectFlags.InstantiationExpressionType) {

```

Comment:
I think having a fallback like this is good; still, I wonder if there's any way you can also special-case the logic here for `typeof`.

```ts
const isInstantiationExpression = !!(symbol.flags & SymbolFlags.Function && getObjectFlags(type) & ObjectFlags.InstantiationExpressionType);
const instantiationExpressionType = type as InstantiationExpressionType;
if (isTypeQueryNode(instantiationExpressionType.node)) {
    // deep clone the type query node
}
```

Reply:
Is ok to emit 

```
foo("");
>foo("") : typeof foo<T>
>foo : <T>(t: T) => typeof foo<T>
```

instead of 

```
foo("");
>foo("") : (t: string) => any
>foo : <T>(t: T) => (t: T) => any
>"" : ""
```
?

Reply:
Personally, I think so - other reviewers might have other thoughts.

Reply:
Oke